src/components/dashboard/expense-list-section/expense-list-section.tsx

```diff
@@ -1,29 +1,72 @@
-import { useState } from "react";
+import React, { useState, useEffect, useCallback } from "react";
 import { SortOrder } from "../../../@types/sortOrderTypes";
+
+import { ExpenseLIstType } from "../../../@types/expense-list-prop";
+import { DataLoader } from "../../shared";
 import ExpenseListToolbar from "./expense-list-toolbar/expense-list-toolbar";
-import { ExpenseList } from "./expense-list";
+import { ExpenseList } from "./expense-list/expense-list";
 import { useExpenseLists } from "./useExpenseLists";
-import { SuspenseComponent } from "../../shared";
 import "./expense-list-section.css";
 
 export const ExpenseListSection = () => {
   const [sortOrder, setSortOrder] = useState<SortOrder>("asc");
-  const { data: section, isLoading, error } = useExpenseLists(0, 5, sortOrder);
+  const [offset, setOffset] = useState(0);
+  const [lists, setLists] = useState<ExpenseLIstType[]>([]);
+  const {
+    data: section,
+    isLoading,
+    error,
+  } = useExpenseLists(offset, 5, sortOrder);
+
+  const updateLists = useCallback(() => {
+    if (section?.data) {
+      if (offset === 0) {
+        setLists(section.data);
+      } else {
+        setLists((prevLists) => [...prevLists, ...section.data]);
+      }
+    }
+  }, [section, offset]);
+
+  useEffect(() => {
```

Comment:
Ofir : Please explain this is not best practice

Reply:
you invoke it when it changes, not when you decide